documentation/modules/exploit/windows/http/plex_unpickle_dict_rce.md

```diff
@@ -0,0 +1,162 @@
+## Vulnerable Application
+
+This module exploits an authenticated Python unsafe `pickle.load` of a
+`Dict` file. An authenticated attacker can create a photo library and
+add arbitrary files to it. After setting the Windows only Plex
+variable `LocalAppDataPath` to the newly created photo library, a file
+named `Dict` will be unpickled, which causes an RCE as the user who
+started Plex. Due to the restrictions of pickle RCE, we write our
+payload to a file then simply execute it through the pickled `Dict`
+file.
+
+If an exploit fails, or is cancelled, `Dict` and `Dict.bat` are
+left on disk, a new `ALBUM_NAME` will be required as subsuquent writes
+will make `Dict-1` and `Dict-1.bat`, and not execute.
+
+A vulnerable version of the software can be downloaded from
+[uptodown.com](https://plex-media-server.en.uptodown.com/windows/versions),
+specifically [1.18.5.2309](https://plex-media-server.en.uptodown.com/windows/download/2177216)
+is vulnerable and used for developing the module.
+
+### Pickle Stub
+
+This exploit requires a python pickle file which can be generated with the following
+code:
+
+```
+import pickle
+import os
+
+class EvilPickle(object):
+  def __init__(self):
+    pass
+  def __reduce__(self):
+     return (os.system,('c:\\Users\\Public\\mememe\\Dict.bat',))
+
+e = EvilPickle()
+pickle.dumps(e)
+"cposix\nsystem\np0\n(S'c:\\\\Users\\\\Public\\\\mememe\\\\Dict.bat'\np1\ntp2\nRp3\n."
+```
+
+However, this was run on Linux and thus `posix` can be seen in the output.  When run
+on Windows, the output will be slightly different and use `nt` isntead of `posix`.
+The Windows version of the code is required.
+
+### Pickle Gotchas
+
+All the examples of Evil Pickle attacks seem to call one command/function.
+[1](https://github.com/fhightower/evil-pickle/blob/master/evil_pickle_writer.py#L17),
+[2](https://medium.com/@abhishek.dev.kumar.94/sour-pickle-insecure-deserialization-with-python-pickle-module-efa812c0d565),
+[3](https://blog.nelhage.com/2011/03/exploiting-pickle/),
+[4](https://davidhamann.de/2020/04/05/exploiting-python-pickle/)
+
+Either due to the pickling, or the HTTP server interaction, or running on Windows, all attempts to either call multi-function such as:
+```eval(base64.b64decode())```
+or execute OS commands with more than one parameter
+```os.system('net user...')```
+or
+```os.system('powershell.exe ...')```
+failed.  With this in mind, the strategy of uploading a payload file, and calling it was the one that worked.
+
+
+## Verification Steps
+
+  1. Install the application
+  2. Register/Connect it to your Plex Premium account
```

Comment:
Pass, not Premium.

Reply:
Are you... reviewing your own PR in GitHub comments?

Reply:
reminders